It seems that Tensorflow_cc does not use the GPU on my installation

I am using a new and clean installation and build of Tensorflow_cc version 2.7 from sources on Ubuntu 20.04 with CUDA 11.6 ( with Nvidia drivers 510, CUDNN 8.3.3.40, etc ). 
All works "fine", including the example provided.
But when running  our application that uses TF , it seems that it does not use the GPU but the CPU. ( according to performance )
My computer has GPU -  NVIDIA GeForce GTX 1650 Ti
We suspect some dependency with CUDA version, or the fact we use previously TF1 graph (but it works ok on other TF2 environment) or something wrong in configurations during build or run.

Can you suggest what can be the reason for this ?
How we can control/monitor if TF uses GPU or CPU on run time ?

Thanks

update: I tried to add path to /usr/lib/gcc/x86_64-linux-gnu/9/cc1plus
before that, I have used a PATH without NVCC so CUDA was disabled and everything was compiled and build ok, including example.cpp which runs without GPU.
Now, using path that includes both NVCC and cc1plus, I got the following error, during build - 
Looks as comptability issue with gcc and cc1plus ?

ERROR: /home/shmulik/.cache/bazel/_bazel_shmulik/4ba6fc284e9b23cd0a456fff9970072d/external/com_google_protobuf/BUILD:301:11: C++ compilation of rule '@com_google_protobuf//:protoc_lib' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc -MD -MF bazel-out/k8-opt/bin/external/com_google_protobuf/_objs/protoc_lib/csharp_repeated_primitive_field.d ... (remaining 48 argument(s) skipped)
cc1plus: warning: unrecognized gcc debugging option: u
cc1plus: warning: unrecognized gcc debugging option: m
cc1plus: warning: unrecognized gcc debugging option: b
cc1plus: warning: unrecognized gcc debugging option: s
cc1plus: warning: unrecognized gcc debugging option: e
cc1plus: warning: unrecognized gcc debugging option: -
cc1plus: warning: unrecognized gcc debugging option: e
cc1plus: warning: unrecognized gcc debugging option: t
cc1plus: error: too many filenames given.  Type cc1plus --help for usage
INFO: Elapsed time: 22.449s, Critical Path: 0.97s
INFO: 112 processes: 96 internal, 16 local.
FAILED: Build did NOT complete successfully
make[2]: *** [CMakeFiles/tensorflow_base.dir/build.make:119: tensorflow-stamp/tensorflow_base-build] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/tensorflow_base.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

--- a/tensorflow_cc/cmake/build_tensorflow.sh.in
+++ b/tensorflow_cc/cmake/build_tensorflow.sh.in
@@ -58,8 +58,8 @@ if [ "$cuda_allowed" == true ] && [ "$cuda_available" == true ]; then
     # according to the above, we should be specifying CUDA compute capabilities as 'sm_XX' or 'compute_XX' from now on
     # add latest PTX for future compatibility
     export TF_CUDA_COMPUTE_CAPABILITIES=${TF_CUDA_COMPUTE_CAPABILITIES:-"sm_52,sm_53,sm_60,sm_61,sm_62,sm_70,sm_72,sm_75,sm_80,sm_86,compute_86"}
-    export TF_CUDA_PATHS=${TF_CUDA_PATHS:-"/opt/cuda-${TF_CUDA_VERSION},/opt/cuda,/usr/local/cuda-${TF_CUDA_VERSION},/usr/local/cuda,/usr/local,/usr/cuda-${TF_CUDA_VERSION},/usr/cuda,/usr"}
     export TF_CUDA_VERSION="$(nvcc --version | sed -n 's/^.*release \(.*\),.*/\1/p')"
+    export TF_CUDA_PATHS=${TF_CUDA_PATHS:-"/opt/cuda-${TF_CUDA_VERSION},/opt/cuda,/usr/local/cuda-${TF_CUDA_VERSION},/usr/local/cuda,/usr/local,/usr/cuda-${TF_CUDA_VERSION},/usr/cuda,/usr"}
     export TF_NCCL_VERSION="$(find /opt /usr -name 'libnccl.so.*' -path '*/cuda*' | tail -n1 | sed -r 's/^.*\.so\.//')"
     export TF_CUDNN_VERSION="$(find /opt /usr -name 'libcudnn.so.*' -path '*/cuda*' | tail -n1 | sed -r 's/^.*\.so\.//')"
 

--- a/tensorflow_cc/cmake/build_tensorflow.sh.in
+++ b/tensorflow_cc/cmake/build_tensorflow.sh.in
@@ -58,7 +58,7 @@ if [ "$cuda_allowed" == true ] && [ "$cuda_available" == true ]; then
     # according to the above, we should be specifying CUDA compute capabilities as 'sm_XX' or 'compute_XX' from now on
     # add latest PTX for future compatibility
     export TF_CUDA_COMPUTE_CAPABILITIES=${TF_CUDA_COMPUTE_CAPABILITIES:-"sm_52,sm_53,sm_60,sm_61,sm_62,sm_70,sm_72,sm_75,sm_80,sm_86,compute_86"}
-    export TF_CUDA_PATHS=${TF_CUDA_PATHS:-"/opt/cuda,/usr/local/cuda,/usr/local,/usr/cuda,/usr"}
+    export TF_CUDA_PATHS=${TF_CUDA_PATHS:-"/opt/cuda-${TF_CUDA_VERSION},/opt/cuda,/usr/local/cuda-${TF_CUDA_VERSION},/usr/local/cuda,/usr/local,/usr/cuda-${TF_CUDA_VERSION},/usr/cuda,/usr"}
     export TF_CUDA_VERSION="$(nvcc --version | sed -n 's/^.*release \(.*\),.*/\1/p')"
     export TF_NCCL_VERSION="$(find /opt /usr -name 'libnccl.so.*' -path '*/cuda*' | tail -n1 | sed -r 's/^.*\.so\.//')"
     export TF_CUDNN_VERSION="$(find /opt /usr -name 'libcudnn.so.*' -path '*/cuda*' | tail -n1 | sed -r 's/^.*\.so\.//')"

